python/tests/expression/test_literals.py

```diff
@@ -0,0 +1,665 @@
+
+    assert timestamp_lit.value == long_lit.value
+
+
+@pytest.mark.parametrize(
+    "decimalType, decimalValue", [(DecimalType(9, 0), "34"), (DecimalType(9, 2), "34.00"), (DecimalType(9, 4), "34.0000")]
+)
+def test_long_to_decimal_conversion(decimalType, decimalValue):
+    lit = literal(34).to(LongType())
+
+    assert lit.to(decimalType).value.as_tuple() == Decimal(decimalValue).as_tuple()
+
+
+def test_float_to_double():
+    lit = literal(34.56)
+    dbl_lit = lit.to(DoubleType())
+
+    assert math.isclose(lit.value, dbl_lit.value)
+
+
+@pytest.mark.parametrize(
+    "decimalType, decimalValue", [(DecimalType(9, 1), "34.6"), (DecimalType(9, 2), "34.56"), (DecimalType(9, 4), "34.5600")]
+)
+def test_float_to_decimal_conversion(decimalType, decimalValue):
+    lit = literal(34.56)
+
+    assert lit.to(decimalType).value.as_tuple() == Decimal(decimalValue).as_tuple()
+
+
+def test_double_to_float_within_bound():
+    lit = literal(34.56).to(DoubleType())
+    float_lit = lit.to(FloatType())
+
+    assert math.isclose(lit.value, float_lit.value)
+
+
+def test_double_to_float_outside_bound():
+    big_lit = literal(FloatType.max + 1.0e37).to(DoubleType())
+    above_max_lit = big_lit.to(FloatType())
+    assert above_max_lit == AboveMax()
+
+    small_lit = literal(FloatType.min - 1.0e37).to(DoubleType())
+    below_min_lit = small_lit.to(FloatType())
+    assert below_min_lit == BelowMin()
+
+
+@pytest.mark.parametrize(
+    "decimalType, decimalValue", [(DecimalType(9, 1), "34.6"), (DecimalType(9, 2), "34.56"), (DecimalType(9, 4), "34.5600")]
+)
+def test_double_to_decimal_conversion(decimalType, decimalValue):
+    lit = literal(34.56).to(DoubleType())
+
+    assert lit.to(decimalType).value.as_tuple() == Decimal(decimalValue).as_tuple()
+
+
+def test_decimal_to_decimal_conversion():
+    lit = literal(Decimal("34.11").quantize(Decimal(".01")))
+
+    assert lit.value.as_tuple() == lit.to(DecimalType(9, 2)).value.as_tuple()
+    assert lit.value.as_tuple() == lit.to(DecimalType(11, 2)).value.as_tuple()
+    assert lit.to(DecimalType(9, 0)) is None
+    assert lit.to(DecimalType(9, 1)) is None
+    assert lit.to(DecimalType(9, 3)) is None
+
+
+def test_timestamp_to_date():
+    epoch_lit = TimestampLiteral(EPOCH.timestamp() * 1000_000)
+    date_lit = epoch_lit.to(DateType())
+
+    assert date_lit.value == 0
+
+
+# STRING
+
+
+def test_string_literal():
+    sqrt2 = literal("1.414").to(StringType())
+    phi = literal("3.141").to(StringType())
+    phi_string_lit = StringLiteral("3.141")
+    phi_double_lit = literal(3.141).to(DoubleType())
+
+    assert sqrt2 != phi
+    assert phi != phi_double_lit
+    assert phi == phi_string_lit
+    assert phi == phi
+    assert sqrt2 < phi
+    assert sqrt2 <= phi
+    assert phi > sqrt2
+    assert phi >= sqrt2
+    assert str(phi) == '"3.141"'
+
+
+def test_string_to_string_literal():
+    assert literal("abc") == literal("abc").to(StringType())
+
+
+def test_string_to_date_literal():
```

Comment:
These tests should use static values from the spec and Java implementation rather than pulling in fastavro. I know that's what the Java implementation does, but that was to build confidence initially. We don't need to do that here and that removes the need for an extra dependency.